pkg/daemons/executor/embed.go

```diff
@@ -47,6 +51,26 @@ func init() {
 
 func (e *Embedded) Bootstrap(ctx context.Context, nodeConfig *daemonconfig.Node, cfg cmds.Agent) error {
 	e.nodeConfig = nodeConfig
+
+	go func() {
+		// Ensure that the log verbosity remains set to the configured level by resetting it at 1-second intervals
+		// for the first 2 minutes that K3s is starting up. This is necessary because each of the Kubernetes
+		// components will initialize klog and reset the verbosity flag when they are starting.
+		logCtx, cancel := context.WithTimeout(ctx, time.Second*120)
+		defer cancel()
```

Comment:
You don't need to explicitly cancel this, it will be cancelled when the timeout expires, or when the parent context is cancelled - either of which are the only way that this function will ever exit and trigger the deferred cancel.
```suggestion
		logCtx, _l := context.WithTimeout(ctx, time.Second*120)
```

Reply:
I agree but not using the cancel function creates a warning so I wanted to avoid that.

Reply:
in fact I had to keep the cancel function, otherwise the linter was making the build failed: https://drone-pr.k3s.io/k3s-io/k3s/6194/1/3

Reply:
ugh stupid linter thinks it knows better. OK.